app/controllers/nonsipp/landorpropertydisposal/landorpropertydisposal.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.nonsipp
+
+import play.api.mvc.Call
+import config.RefinedTypes.{Max50, Max5000}
+import models.SchemeId.Srn
+import pages.nonsipp.landorpropertydisposal.LandOrPropertyDisposalProgress
+import models.UserAnswers
+import utils.FunctionKUtils._
+import viewmodels.models.SectionJourneyStatus
+
+import scala.concurrent.Future
+
+package object landorpropertydisposal {
+
+  implicit class CallOps(call: Call) {
+
+    val isCyaPage: Boolean = {
+      val pattern =
+        s"^\\/pension-scheme-return\\/[^\\/]+\\/(change-check-answers|check-answers)-land-property-disposal\\/(?!.*\\/\\d+\\/\\d+$$).*"
```

Comment:
Should this be `(change|check-answers)` instead of `(change-check-answers|check-answers)`?

Reply:
This is fine, it can also be written as `(change-)?check-answers`

Reply:
![image](https://github.com/user-attachments/assets/ffe15ea7-6a7c-4291-84b3-9e6d582019f8)

This is how we have it in the router

Reply:
Yes, you're right, please ignore.